# HG changeset patch
# User Darren Salt <devspam@moreofthesa.me.uk>
# Date 1394989339 0
#      Sun Mar 16 17:02:19 2014 +0000
# Node ID 702c252f32002c167b1755bd0a6f159f511fa271
# Parent  8c0acbb8fcbb765120eb31ca637ecf82a1f213e4
Fix another UI-related crash on exit.

diff -r 8c0acbb8fcbb -r 702c252f3200 src/xitk/main.c
--- a/src/xitk/main.c	Sun Mar 16 16:59:38 2014 +0000
+++ b/src/xitk/main.c	Sun Mar 16 17:02:19 2014 +0000
@@ -1255,8 +1255,10 @@
       gGui->mixer.volume_level = (aevent->left + aevent->right) / 2;
       if(gGui->mixer.method == SOUND_CARD_MIXER) {
 	gGui->mixer.mute = aevent->mute;
-	xitk_slider_set_pos(panel->mixer.slider, gGui->mixer.volume_level);
-	xitk_checkbox_set_state(panel->mixer.mute, gGui->mixer.mute);
+	if (panel) {
+	  xitk_slider_set_pos(panel->mixer.slider, gGui->mixer.volume_level);
+	  xitk_checkbox_set_state(panel->mixer.mute, gGui->mixer.mute);
+	}
       }
     }
     break;
